File: android/app/src/main/java/com/snappet/mobile/feature/nutrition/NutritionBudget.kt
package com.snappet.mobile.feature.nutrition

/**
 * Pure calorie budget snapshot — no Android / Compose imports so it is testable on any JVM host.
 * Mirrors iOS `NutritionBudget` struct.
 */
data class NutritionBudget(
    val consumedKcal: Double,
    val goalKcal: Int,
    val activeBurnedKcal: Double = 0.0,
) {
    val remainingKcal: Double get() = goalKcal + activeBurnedKcal - consumedKcal
    val netKcal: Double get() = consumedKcal - activeBurnedKcal
    val ringFraction: Float get() =
        if (goalKcal == 0) 0f else (consumedKcal / goalKcal).coerceIn(0.0, 1.0).toFloat()

    companion object {
        /** Mifflin–St Jeor BMR (kcal/day). Used as HealthKit active-burn fallback. */
        fun mifflinBmr(weightKg: Double, heightCm: Double, ageYears: Int, isMale: Boolean): Double {
            val base = 10 * weightKg + 6.25 * heightCm - 5 * ageYears
            return if (isMale) base + 5 else base - 161
        }
    }
}

File: ios/App/Snappet/Features/Nutrition/NutritionBudget.swift
import Foundation

/// Calorie budget state for today. Pure value type — no device imports.
struct NutritionBudget {
    let consumed: Double
    let goal: Double
    let activeBurned: Double

    var remaining: Double { goal - consumed }
    var net: Double { consumed - activeBurned }
    var ringFraction: Double { min(max(consumed / max(goal, 1), 0), 1) }

    /// Mifflin–St Jeor BMR fallback when HealthKit energy is unavailable.
    static func mifflinBMR(weightKg: Double, heightCm: Double, ageYears: Int, isMale: Bool) -> Double {
        let base = 10 * weightKg + 6.25 * heightCm - 5 * Double(ageYears)
        return isMale ? base + 5 : base - 161
    }
}
